xchange-cexio/src/main/java/info/bitrich/xchangestream/cexio/CexioStreamingRawService.java

```diff
@@ -23,44 +30,139 @@
     public static final String PONG = "pong";
     public static final String ORDER = "order";
     public static final String TRANSACTION = "tx";
+    public static final String ORDERBOOK = "order-book-subscribe";
+    public static final String ORDERBOOK_UPDATE = "md_update";
 
     private String apiKey;
     private String apiSecret;
-
+    private boolean isAuthenticated = false;
+    private final Lock authLock = new ReentrantLock();
+    private final Condition authCond = authLock.newCondition();
+    private final boolean isTest;
+    
     private PublishSubject<Order> subjectOrder = PublishSubject.create();
     private PublishSubject<CexioWebSocketTransaction> subjectTransaction = PublishSubject.create();
-
-    public CexioStreamingRawService(String apiUrl) {
+    
+    public CexioStreamingRawService(String apiUrl, boolean isTest) {
         super(apiUrl, Integer.MAX_VALUE);
+        this.isTest = isTest;
     }
-
+    
+    public void waitForAuthIfNeeded() {
+    	authLock.lock();
+    	try
+    	{
+    		while (!isAuthenticated && !isTest)
```

Comment:
Could you remove the `isTest` argument and simply use `apiKey != null`?  That is, authenticate if you have authentication details, and don't authenticate if you don't have them (matches other exchanges - see #274)

Reply:
Done in d1c7ab91daaa8867f279993a72e812c2e3dd4e90

Reply:
We do not need an isTest argument for the existing test suite because none of the tests connect to the exchange, so bypass waiting on the returned completable.